gef-ui/server/controllers/about-facility/index.validate-and-update-about-facility.test.js

```diff
@@ -826,5 +806,63 @@ describe('validateAndUpdateAboutFacility', () => {
         }),
       );
     });
+
+    it('redirects user to facility end date page if using facility end date', async () => {
+      mockRequest.body.facilityType = CONSTANTS.FACILITY_TYPE.CASH;
+      mockRequest.body.hasBeenIssued = 'true';
+      mockRequest.body.facilityName = 'Name';
+      mockRequest.body.shouldCoverStartOnSubmission = 'true';
+      mockRequest.body.hasBeenIssued = 'true';
+      mockRequest.body.monthsOfCover = '10';
+      mockRequest.body['cover-end-date-day'] = format(threeMonthsAndOneDayFromNow, 'd');
+      mockRequest.body['cover-end-date-month'] = format(threeMonthsAndOneDayFromNow, 'M');
+      mockRequest.body['cover-end-date-year'] = format(threeMonthsAndOneDayFromNow, 'yyyy');
+
+      mockRequest.body.isUsingFacilityEndDate = 'true';
+
+      await validateAndUpdateAboutFacility(mockRequest, mockResponse);
+
+      expect(mockResponse.redirect).toHaveBeenCalledWith('/gef/application-details/123/facilities/xyz/facility-end-date');
+    });
+
+    it('redirects user to bank review date page if not using facility end date', async () => {
+      mockRequest.body.facilityType = CONSTANTS.FACILITY_TYPE.CASH;
+      mockRequest.body.hasBeenIssued = 'true';
+      mockRequest.body.facilityName = 'Name';
+      mockRequest.body.shouldCoverStartOnSubmission = 'true';
+      mockRequest.body.hasBeenIssued = 'true';
+      mockRequest.body.monthsOfCover = '10';
+      mockRequest.body['cover-end-date-day'] = format(threeMonthsAndOneDayFromNow, 'd');
+      mockRequest.body['cover-end-date-month'] = format(threeMonthsAndOneDayFromNow, 'M');
+      mockRequest.body['cover-end-date-year'] = format(threeMonthsAndOneDayFromNow, 'yyyy');
+
+      mockRequest.body.isUsingFacilityEndDate = 'false';
+
+      await validateAndUpdateAboutFacility(mockRequest, mockResponse);
+
+      expect(mockResponse.redirect).toHaveBeenCalledWith('/gef/application-details/123/facilities/xyz/bank-review-date');
```

Comment:
Sorry -- Again because I'm doing this late at night -- What happens if we dont select anything? Do we have a default value? Does it return to the same page but with errors?

Reply:
The test case above
```
    it('shows error message if isUsingFacilityEndDate is not a boolean', async () => {
```
will validate that `isUsingFacilityEndDate` is either true or false